File: Java_MVVM/app/src/main/java/eu/gotoinc/requesinjava_mvvm/repository/RetrofitRequest.java
package eu.gotoinc.requesinjava_mvvm.repository;


import java.util.List;

import eu.gotoinc.requesinjava_mvvm.repository.data_models.responses.PaginationResponse;
import eu.gotoinc.requesinjava_mvvm.repository.data_models.responses.UserResponse;
import io.reactivex.Observable;
import io.reactivex.Single;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RetrofitRequest {
    @GET(ApiConsts.API_VERSION + ApiConsts.LIST_OF_USERS)
    Single<PaginationResponse<List<UserResponse>>> getUsersList(@Query("page")int page);
}

File: Kotlin_MVVM/app/src/main/java/eu/gotoinc/kotlinmvvm/repository/RetrofitRequest.kt
package eu.gotoinc.kotlinmvvm.repository

import eu.gotoinc.kotlinmvvm.repository.datamodel.response.PaginationResponse
import eu.gotoinc.kotlinmvvm.repository.datamodel.response.UserResponse
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface RetrofitRequest {
    @GET(API_VERSION + LIST_OF_USERS)
    fun getUsersList(@Query("page") page: Int): Single<PaginationResponse<List<UserResponse>>>
}
